### ISSUE TITLE
Non-US keyboard - pound / hash symbol (#)

### ISSUE THREAD

**Issue by andypiper** [dupontgu/dial_toner#1]
I'm using a v3 Dial Toner connected to macOS Ventura.

When I hit the button in the default hex mode, I'm getting (for example) `£E4512A` <- generated directly from the key pad right now.

The `£` symbol should be a `#` symbol 🙂 

I'm not 100% sure how to fix this in the code.

On my keyboard, pressing the 3 key with Shift -> `£` and with Alt -> `#`
Nevertheless I was kind of surprised that directly sending the string `#E4512A` (generated via `config.py`) is getting converted to `£E4512A`

- looked into the `KeyboardLayout` for CircuitPython, but I'm not finding (for example) a British English option. It looks like (?) the `KeyboardLayout` assumes specific key locations? Should I add a UK English layout in 
- looked at ways to send a `POUND` keycode (considered removing `#` from `hex_sequence` in `config.py` and having the symbol inserted in `dt_keyboard.py` or, having `dt_keyboard.py` do some kind of filtering to replace `#` in `str` if it is passed in).
- basically I've tried a number of things but seem to be struggling to figure out the "best" way to get the actual `#` character sent to the computer.

It might be nice if this could somehow be a config option on the 5th dial or something, but a code change for the end-user is also OK...

**Comment by dupontgu** [comment]
Hi Andy, thanks for reporting and looking into it! I can't believe I fell into this trap again, I had a similar issue on my T9 project when I went to intro multiple languages. 

Did you have any luck with your second approach? Since there are really no other language-specific components to this project. Would be great if it could just be universal. From a _product_ standpoint, that would be the best option in my opinion.

If not, then yeah - A British English `KeyboardLayout` would be the way to go. I can look into that if you haven't already gotten started. And I like the 5th dial config idea, though it doesn't scale particularly well. Probably not a problem given the scale of this operation. At the very least, I'll rewrite the code to autodetect a custom `KeyboardLayout` file on the CircuitPython drive so no code needs to be altered by the user.

**Comment by andypiper** [comment]
Totally recognise that it's a niche/edge case related to internationalisation, so the settings doesn't scale brilliantly.

So I tried a few approaches, either `POUND` or something like

```
keyboard.press(Keycode.LEFT_ALT)
keyboard.press(Keycode.KEYPAD_THREE)
keyboard.press(Keycode.KEYPAD_FIVE)
keyboard.release_all()
```

... right before the `keyboard_layout.send()` but get nothing out at all from the button, until I revert to your code (and I don't see a way to see any error output - I'm typically more of a MicroPython person though). I might have another go at this at the weekend but not immediately able to play further.

Lovely little device, thanks! 👍🏻

**Comment by dupontgu** [comment]
Ok - I'm having luck with:
```
self.keyboard.press(Keycode.LEFT_ALT)
self.keyboard.press(Keycode.THREE) #Note - not the KEYPAD_THREE, apparently there's a difference
time.sleep(.02)
self.keyboard.release_all()
```
on a Mac, set to British input source.

I think I'm going to do the 5th dial config idea with 3 different modes:
1. US (as written now, just use the "#ffffff" string as-is)
2. UK (Keycode.LEFT_ALT + THREE, then string)
3. INTL (POUND, then string)

^ seems like a few other languages use POUND for #, so that'll be a catchall for now.

--------

I'm conveniently readying a release right now as I prepare to release the "Ultra" version of the dial toner. There weren't going to be any new features for the v3 Dial Toner but this will make it's way in! I will post back here when I have a release candidate of the CircuitPython bundle.

**Comment by dupontgu** [comment]
@andypiper if you want to give the updated code a test, I would greatly appreciate it. The `DT Basic Bundle` for the latest is available here: https://github.com/dupontgu/dial_toner/actions/runs/3381251966

Quite a few internal changes, so I would just overwrite all the files you have.
The `keyboard_mode` parameter is on the 5th knob when you go into config mode.

0 -> US keyboard
1 -> UK
2 -> INTL (everybody else?)
else -> defaults to 0

**Comment by andypiper** [comment]
Nice! working well here.

(I just renamed my CIRCUITPY drive as well, since I also have a macropad with the same name...)

Thanks for the work on my regional edge-case 😄

**Comment by dupontgu** [comment]
merged in #2!
